### PR TITLE
Update poll break condition in pod status check in the test extensions

### PR DESCRIPTION
**Pull request by caliskanugur**
## Problem

Currently, the tests that are using `workloads/pods.StatusPods` (https://github.com/rancher/rancher/blob/release/v2.7/tests/framework/extensions/workloads/pods/pod_status.go#L49) are flaky because of two separate issues:

- Default true makes the check happen for once, and pods might not be in the expected conditions and cause to send empty slices to the tests where they are expected not to be empty, with this error: `Should NOT be empty, but was []`
- If the container statuses aren't initialized given time, indexing panics with this error: `Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)`
 
## Solution

- Prevent poll exit by default to false
- Make errors and results length check, and If the results slice is greater than zero and if the error slice is zero, break the poll with true. 
- For the container statuses, return false by default if there's no item to retrigger the pod call

### Automated Testing

* Test types added/modified:
    * Integration (Go Framework)
    * Validation (Go Framework)

### CODE REVIEW

**Review by markusewalker — APPROVED** [review]
LGTM

File: tests/framework/extensions/workloads/pods/pod_status.go [hunk 1]
```diff
@@ -46,18 +46,30 @@ func StatusPods(client *rancher.Client, clusterID string) ([]string, []error) {
...
+			}
+
 			image := podStatus.ContainerStatuses[0].Image
 			phase := podStatus.Phase
+
 			if phase == corev1.PodFailed || phase == corev1.PodUnknown {
 				podErrors = append(podErrors, fmt.Errorf("ERROR: %s: %s", pod.Name, podStatus))
 				logrus.Infof("Pod %s: Not active | Image %s", pod.Name, image)
 				return false, nil
-			} else if phase == corev1.PodRunning {
```
Comment (igomez06) [inline tests/framework/extensions/workloads/pods/pod_status.go:55]: Why did you change the else if?

**Review by igomez06 — COMMENTED** [review]
one question everything else looks good

File: tests/framework/extensions/workloads/pods/pod_status.go (hunk 1 above)
Comment (caliskanugur) [reply tests/framework/extensions/workloads/pods/pod_status.go:55]: Mainly personal preference; I find if there are a few conditions, it's more readable with multiple ifs rather than else ifs. I can change it, surely. Would you like me to change it back?

Comment (igomez06) [reply tests/framework/extensions/workloads/pods/pod_status.go:55]: I was trying to save checks but with the return it does that anyways. So ignore me lol

**Review by igomez06 — APPROVED** [review]
LGTM
